### ISSUE TITLE
Default podman-machine-default on Windows missing default `seccomp.json`?

### ISSUE THREAD

**Issue by rdbisme** [containers/podman#24318]
### Issue Description

I use `podman` from my WSL (Arch) using the shared socket mount + `podman-remote`. I was experimenting today with some `molecule` stuff, and the build was failing with an error related to a missing `/etc/containers/seccomp.json`. 

I copied the default `seccomp.json` from my arch WSL to `/etc/containers` of the podman-machine-default and things seem to work.

Is it expected that the `seccomp.json` is missing?

### Steps to reproduce the issue

Steps to reproduce the issue
1. Run a custom build with molecule which builds a custom `archlinux:latest` image `molecule create`

### Describe the results you received

opening seccomp profile failed: open /etc/containers/seccomp.json: no such file or directory

### Describe the results you expected

`/etc/containers/seccomp.json` is available in the WSL VM used by podman?

### podman info output

host:
  arch: amd64
  buildahVersion: 1.36.0
  cgroupControllers:
  - cpu
  - memory
  - pids
  cgroupManager: cgroupfs
  cgroupVersion: v2
  conmon:
    package: conmon-2.1.10-1.fc40.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.10, commit: '
  cpuUtilization:
    idlePercent: 99.16
    systemPercent: 0.34
    userPercent: 0.5
  cpus: 12
  databaseBackend: sqlite
  distribution:
    distribution: fedora
    variant: container
    version: "40"
  eventLogger: journald
  freeLocks: 2048
  hostname: CiD
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 524288
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 524288
      size: 65536
  kernel: 5.15.153.1-microsoft-standard-WSL2
  linkmode: dynamic
  logDriver: journald
  memFree: 13978820608
  memTotal: 16732319744
  networkBackend: netavark
  networkBackendInfo:
    backend: netavark
    dns:
      package: aardvark-dns-1.11.0-1.fc40.x86_64
      path: /usr/libexec/podman/aardvark-dns
      version: aardvark-dns 1.11.0
    package: netavark-1.11.0-1.fc40.x86_64
    path: /usr/libexec/podman/netavark
    version: netavark 1.11.0
  ociRuntime:
    name: crun
    package: crun-1.15-1.fc40.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 1.15
      commit: e6eacaf4034e84185fd8780ac9262bbf57082278
      rundir: /run/user/1000/crun
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +LIBKRUN +WASM:wasmedge +YAJL
  os: linux
  pasta:
    executable: /usr/bin/pasta
    package: passt-0^20240624.g1ee2eca-1.fc40.x86_64
    version: |
      pasta 0^20240624.g1ee2eca-1.fc40.x86_64
      Copyright Red Hat
      GNU General Public License, version 2 or later
        <https://www.gnu.org/licenses/old-licenses/gpl-2.0.html>
      This is free software: you are free to change and redistribute it.
      There is NO WARRANTY, to the extent permitted by law.
  remoteSocket:
    exists: true
    path: /run/user/1000/podman/podman.sock
  rootlessNetworkCmd: pasta
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /etc/containers/seccomp.json
    selinuxEnabled: false
  serviceIsRemote: true
  slirp4netns:
    executable: ""
    package: ""
    version: ""
  swapFree: 4294967296
  swapTotal: 4294967296
  uptime: 2h 14m 18.00s (Approximately 0.08 days)
  variant: ""
plugins:
  authorization: null
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  - ipvlan
  volume:
  - local
registries:
  search:
  - docker.io
store:
  configFile: /home/user/.config/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /home/user/.local/share/containers/storage
  graphRootAllocated: 1081101176832
  graphRootUsed: 4447715328
  graphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Supports shifting: "false"
    Supports volatile: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 2
  runRoot: /run/user/1000/containers
  transientStore: false
  volumePath: /home/user/.local/share/containers/storage/volumes
version:
  APIVersion: 5.1.1
  Built: 1717459200
  BuiltTime: Tue Jun  4 02:00:00 2024
  GitCommit: ""
  GoVersion: go1.22.3
  Os: linux
  OsArch: linux/amd64
  Version: 5.1.1

### Podman in a container

No

### Privileged Or Rootless

Rootless

### Upstream Latest Release

Yes

### Additional environment details

Running `molecule create` which runs `podman-remote` via `ansible` on an Archlinux WSL

### Additional information

Additional information like issue happens only occasionally or issue happens with a particular architecture or on a particular setting

**Comment by Luap99** [comment]
This sounds like a bug podman-remote should not read `/etc/containers/seccomp.json` on the client

**Comment by rhatdan** [comment]
Yes I agree the default seccomp.json file should only be read inside of the Machine.   One issue might be how well we handle

podman run --security-opt seccomp:/tmp/myseccomp.json ...

In this case the myseccomp.json should be read from the client and sent to the host.

**Comment by rdbisme** [comment]
> This sounds like a bug podman-remote should not read /etc/containers/seccomp.json on the client
Not sure I'm understanding, but I think that's what's happening? The problem is that in `podman-remote-default:/etc/containers` there's no `seccomp.json`

**Comment by rdbisme** [comment]
Hello, I just tried to build on another Windows PC and it reproduces the problem.

```
❯ podman-remote build  /home/rdb/.cache/molecule/extensions/default/ -f /home/rdb/.cache/molecule/extensions/default/Dockerfile_archlinux_latest
STEP 1/2: FROM archlinux:latest
STEP 2/2: RUN pacman-key --init && pacman -Syu --noconfirm && pacman -S --noconfirm python readline
Error: building at STEP "RUN pacman-key --init && pacman -Syu --noconfirm && pacman -S --noconfirm python readline": opening seccomp profile failed: open /etc/containers/seccomp.json: no such file or directory
```

Is it a bug? I think it is...

**Comment by Luap99** [comment]
Please update your podman version on the server, i.e. recreate the podman machine or run dnf update in the machine.

I read the code on main and do not see that we ever try to read the path on the client so I think that works fine. Is this specific to podman build? Because I only tested podman run.

**Comment by Luap99** [comment]
yeah ok I think the issue is in the buildah code. The client side there checks /etc/containers/seccomp.json and /usr/share/containers/seccomp.json and then sends this path to the server. The client should not send such defaults to the server as the server must use its own defaults

**Comment by rdbisme** [comment]
On the machine: 

```
[user@Baaj ~]$ podman --version
podman version 5.2.3
```

and indeed it's specific to `podman build`.

**Comment by rdbisme** [comment]
Just a note that this is still a problem. I configured the same thing brand new on a pc and that doesn't work with the same error.

**Comment by DelusionalLogic** [comment]
The problem is still present in 5.6.0. I fixed it by copying the default over myself.

    cp /usr/share/containers/seccomp.json /etc/containers/seccomp.json
